openpgp/packet/packet.go

```diff
@@ -395,7 +395,9 @@ func Read(r io.Reader) (p Packet, err error) {
 		err = p.parse(contents)
 	}
 	if err != nil {
-		consumeAll(contents)
+		if _, errConsume := consumeAll(contents); errConsume != nil {
+			err = errConsume
+		}
 	}
```

Comment:
This missing error check revealed that there was a broken key being used for tests. With this check, `TestKeyExpiry` fails on invalid signature data. The test is omitted until a correct key is supplied (see `keys_test.go` and `keys_test_data.go`)

Reply:
Replaced broken key -- thanks to @twiss for providing a correct key Hex